### PR TITLE
Make the project usable as an npm dependency

### PR DESCRIPTION
**Pull request by dmisdm**
#### This issue is an enhancement request
It would be great if this project could be more easily used as an NPM dependency in other applications.

Understand this may be a big change but i thought i may as well propose it to gauge reaction.

What i've done:
- Produce build using webpack
- Update package.json with main field
- Added readium-shared-js as a dependency instead of a submodule
 - Currently referring to my own repo fork. There is a PR for the same thing on readium-shared-js

### CODE REVIEW

**Comment by danielweck** [comment]
Pretty much the same comments are with the other PR:
https://github.com/readium/readium-shared-js/pull/451
(so I won't duplicate them here)

File: js/Readium.js [hunk 1]
```diff
@@ -12,10 +12,10 @@
 //  prior written permission.
 
 
-define(['readium_shared_js/globals', 'text!version.json', 'jquery', 'underscore', 'readium_shared_js/views/reader_view', 'readium_js/epub-fetch/publication_fetcher',
```
Comment (danielweck) [inline js/Readium.js]: Does this work with all AMD loaders? (RequireJS and Almond are the ones used in "multiple" and "single" bundle builds, respectively)

File: js/epub-fetch/zip_resource_fetcher.js [hunk 2]
```diff
@@ -11,7 +11,7 @@
 //  used to endorse or promote products derived from this software without specific
 //  prior written permission.
 
-define(['jquery', 'URIjs', './discover_content_type', 'zip-ext', 'readium_shared_js/helpers'], function ($, URI, ContentTypeDiscovery, zip, Helpers) {
```
Comment (danielweck) [inline js/epub-fetch/zip_resource_fetcher.js]: Why is `zip` removed? This should not be a global, it should be injected via `zip-ext` (which as the name implies, provides the necessary extensions).

File: yarn-error.log [hunk 3]
```diff
@@ -0,0 +1,7046 @@
+Arguments:
```
Comment (danielweck) [inline yarn-error.log]: This file should not be committed / pushed to Git.

**Comment by fgilio** [comment]
Heys guys are this and this https://github.com/readium/readium-shared-js/pull/451 PR still alive?

**Comment by fgilio** [comment]
Thanks for the in depth explanation @danielweck!
We're currently using this https://www.npmjs.com/package/@evidentpoint/readium-js, and while it works I thinks it's a really old build. I tried to replicate it building from this repo source... but had no success.
I think I'll try with your instructions :) and use the lib in that way, instead of with @evidentpoint/readium-js

**Comment by danielweck** [comment]
Guessing:
https://www.npmjs.com/package/@evidentpoint/readium-js
... was published from:
https://github.com/evidentpoint/readium-js/
?
Not sure. Probably best to ask @jccr or @roberthodan

**Comment by fgilio** [comment]
Well the thing is that https://www.npmjs.com/package/@evidentpoint/readium-js points to *this* repo. But it could all be a huge misunderstanding.

**Pull request closed without merging**